lib/syskit/task_context.rb

```diff
@@ -1151,7 +1200,7 @@ def queue_last_chance_to_stop
 
         # Interrupts the execution of this task context
         event :stop do |_context|
-            interrupt!
+            interrupt! unless read_only?
```

Comment:
Here, you will need to emit stop (unlike for start)

The conflict comes from the meaning of 'stop' in this case. It may have two:
- the remote task stopped
- the Syskit app does not need to watch that remote task anymore

We don't have a way to separate the two here, but I think it does not matter in practice.

Reply:
If I emit stop, the `stop!` command will fail:

https://github.com/rock-core/tools-roby/blob/ff6744e5bd00e4e757df450ef0a62889915d0eaa/lib/roby/task_event_generator.rb#L257

Reply:
> If stop is emitted, the stop! command will be rejected:

If stop is *already* emitted. Why would it be already emitted ?

Reply:
> If stop is _already_ emitted. Why would it be already emitted ?

Sorry, I didn't get your point

Reply:
More  specifically this:
```rb
            it "does not emit interrupted if the task is stopped" do
                task = @task
                Orocos.allow_blocking_calls { task.orocos_task.start(false) }
                execute { task.start! }

                expect_execution { task.stop! }
                    .to { emit task.stop_event }
            end
```

causes: 

"Roby::CommandRejected: stop! called by [] but the task has finished. Task has been terminated by []."

Because when `task,stop!` is called, if it emits stops, the command itself will be rejected due to the way "check_call_validity" is written

Reply:
nevermind, my mistake